diff -ruN libinput-1.11.1/tools/shared.c libinput-1.11.1-segfault/tools/shared.c
--- libinput-1.11.1/tools/shared.c	2018-06-19 13:00:35.000000000 +0800
+++ libinput-1.11.1-segfault/tools/shared.c	2018-07-03 15:34:32.275138223 +0800
@@ -232,7 +232,7 @@
 	if (fd < 0)
 		fprintf(stderr, "Failed to open %s (%s)\n",
 			path, strerror(errno));
-	else if (*grab && ioctl(fd, EVIOCGRAB, (void*)1) == -1)
+	else if (grab && *grab && ioctl(fd, EVIOCGRAB, (void*)1) == -1)
 		fprintf(stderr, "Grab requested, but failed for %s (%s)\n",
 			path, strerror(errno));
 
